### ISSUE TITLE
Wrong initialization sequence in mage.priceBox widget (price-box.js)

### ISSUE THREAD

**Issue by yariksheptykin** [magento/magento2#6117]
During initialization of priceBox [the following code is executed](https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/view/base/web/js/price-box.js#L42-L43):

```
this._setDefaultsFromPriceConfig();
this._setDefaultsFromDataSet();
```

However `_setDefaultsFromPriceConfig` [relies on](https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/view/base/web/js/price-box.js#L203) `this.options.productId`. This value is set in `setDefaultsFromDataSet` [here](https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/view/base/web/js/price-box.js#L180). 

Current sequence of function calls leaves `this.options.productId` always empty in `this._setDefaultsFromPriceConfig()`.

I stumbled upon this issue while trying to add a custom price type and configuring priceBox to render it. With the current sequence of calls priceBox seem to ignore the configuration of the price types.
### Preconditions

No pre-conditions. Just debug the function using browser developer tools.
### Steps to reproduce
1. Create a product which has a price.
2. Navigate to the product details page 
3. Using your JS debugger set the breakpoint on https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/view/base/web/js/price-box.js#L206
### Expected result

1. The line should be executed.
### Actual result

1. This line never gets called.

**Comment by veloraven** [comment]
@yariksheptykin thank you for your report.
Please, provide the used Magento version.

**Comment by yariksheptykin** [comment]
2.1.0

**Comment by yariksheptykin** [comment]
Changing:

```
this._setDefaultsFromPriceConfig();
this._setDefaultsFromDataSet();
```

to

```
this._setDefaultsFromDataSet();
this._setDefaultsFromPriceConfig();
```

Fixes the problem.

**Comment by andimov** [comment]
@yariksheptykin
Thank you for reporting!
I've created an internal ticket MAGETWO-57136 for this issue and linked it to this one.

**Comment by yariksheptykin** [comment]
Awesome! You want me to fix and do a PR?

**Comment by KrystynaKabannyk** [comment]
@yariksheptykin, Closing this issue as it has been fixed under MAGETWO-57136 for 2.1.3.

**Comment by yariksheptykin** [comment]
Awesome. Thanks!
